cmd/ferretdb/main.go

```diff
@@ -519,6 +498,33 @@ func run() {
 		logger.Sugar().Fatalf("Failed to construct listener: %s.", err)
 	}
 
+	if cli.DebugAddr != "" && cli.DebugAddr != "-" {
+		wg.Add(1)
+
+		go func() {
+			defer wg.Done()
+
+			l := logger.Named("debug")
+
+			opts := &debug.ListenOpts{
+				TCPAddr: cli.DebugAddr,
+				L:       l,
+				R:       metricsRegisterer,
+			}
+
+			if cli.Setup.Database != "" {
+				opts.Backend = h.Backend
+			}
+
+			h, err := debug.Listen(opts)
+			if err != nil {
+				l.Sugar().Fatalf("Failed to create debug handler: %s.", err)
+			}
+
+			h.Serve(ctx)
+		}()
+	}
+
```

Comment:
It's possible to not move the whole debug section from the beginning, by calling `h.Serve()`, `l.Run()`, and `r.Run()`, later in the code. If the approach is alright, I'll proceed to do that, for now less changes is better.

Reply:
What is the issue here? Moving the code? :thinking:

Reply:
I believe we'd like to make sure that the debug handler is the first that is running